derive/src/lib.rs

```diff
@@ -252,10 +254,57 @@ fn derive_marker_trait_inner<Trait: Derivable>(
 
     #trait_impl_extras
 
-    unsafe impl #impl_generics #trait_ #trait_params for #name #ty_generics #where_clause {
+    unsafe impl #impl_generics #trait_ for #name #ty_generics #where_clause {
       #trait_impl
     }
 
     #implies_trait
   })
 }
+
+/// Add a trait marker to the generics if it is not already present
+fn add_trait_marker(
+  generics: &mut syn::Generics,
+  trait_name: &syn::Path
+) {
```

Comment:
```rs
fn add_trait_marker(
  generics: &mut syn::Generics,
  trait_name: &syn::Path
) {
    let type_params: Vec<WherePredicate> =
        generics.type_params().map(|p| &p.ident).map(|ty| parse_quote!(
            #ty : #trait_name
        )).collect();
    generics.make_where_clause().predicates.extend(type_params);
}
```

These 5 lines seem more straightforward to read/maintain than your 40 lines heroically bypassing `parse_quote!` (I find the maintainability benefits of `parse_quote!` to outweigh the sub-milli-second runtime-of-the-macro overhead of parsing) 🙃

Reply:
That's definitely a better approach than mine. I've implemented it.